src/test/java/com/maveric/userservice/service/UserServiceTest.java

```diff
@@ -0,0 +1,332 @@
+
+        given(iUserRepository.findByEmail(prepaireUserDto().getEmail())).willReturn(Optional.of(userEntity));
+        when(iUserService.getUserByEmail(prepaireUserDto().getEmail())).thenThrow(new UserException(new Exception(),new Exception().getMessage()));
+        Assertions.assertThrows(UserException.class,() ->iUserService.getUserByEmail(prepaireUserDto().getEmail()));
+
+    }
+    @DisplayName("JUnit test for updateUser method")
+    @Test
+    public void givenUserId_whenUpdateUser_thenReturnObject(){
+        // given - precondition or setup
+        Integer userId = 1;
+
+       // given(iUserRepository.findByEmail(prepaireUserDto().getEmail())).willReturn(Optional.empty());
+        given(iUserRepository.findById(userId)).willReturn(Optional.of(userEntity));
+         userEntity.setPhoneNumber("9420887276");
+        given(iUserRepository.save(userEntity)).willReturn(userEntity);
+
+         ResponseEntity response = iUserService.updateUser(prepaireUserDto());
+         System.out.println(response.getBody());
+        Assertions.assertNotNull(response.getBody());
+        Assertions.assertEquals(HttpStatus.OK,response.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("JUnit test for updateUser method Throw Exception")
+    public  void  givenUser_whenUpdateUser_ThenThrowException(){
+        given(iUserRepository.findById(1)).willReturn(Optional.of(userEntity));
+        userEntity.setPhoneNumber("9420887276");
+        when(iUserRepository.save(userEntity)).thenThrow(new UserException(new Exception(),new  Exception().getMessage()));
+        Assertions.assertThrows(UserException.class,()->iUserService.updateUser(prepaireUserDto()));
+    }
+
+    @Test
+    @DisplayName("JUnit test for updateUser method when for given userId User is not present")
+    public  void  givenUserId_whenUpdateUser_ThenReturnMsgUserNotPresent(){
+        given(iUserRepository.findById(1)).willReturn(Optional.empty());
+        Map<String,String> error=new HashMap();
+        error.put(UserServiceConstant.error_code,UserServiceConstant.BAD_REQUEST);
+        error.put(UserServiceConstant.error_message,UserServiceConstant.USER_IS_NOT_PRESENT_FOR_GIVEN_ID);
+        ResponseEntity response = iUserService.updateUser(prepaireUserDto());
+        Assertions.assertEquals(error,response.getBody());
+    }
+
+    @Test
+    @DisplayName("JUnit test for getUsers method pagination")
+    public  void  whenGetUsers_ThenReturnListUser(){
+        Pageable paging = PageRequest.of(0, 2, Sort.by("inventory"));
+        Page<UserEntity> userEntityPage= Mockito.mock(Page.class);
+        given(iUserRepository.findAll(paging)).willReturn(userEntityPage);
+        ResponseEntity response = iUserService.getUsers(paging);
+        Assertions.assertEquals(new ArrayList<UserEntity>(),response.getBody());
+    }
+
+    @Test
+    @DisplayName("JUnit test for getUsers method pagination return Exception")
+    public  void  whenGetUsers_ThenThrowException(){
```

Comment:
dnt use _ in method names, change all test method names to meaningfull names with onlt camelcasing